### PR TITLE
Adding data to sentry

### PR DESCRIPTION
**Pull request by kostagersh**

### CODE REVIEW

File: packages/corvid-cli/src/utils/electron.js [hunk 1]
```diff
@@ -152,14 +153,29 @@ async function openWindow(app, windowOptions = {}) {
       setTimeout(resolve, isDevTools ? 1000 : 0);
     }).then(async () => {
       const corvidConfig = await readCorvidConfig(".");
-
+      win.webContents
+        .executeJavaScript(
+          "(() => ({santaBase: window.santaBase, editorBase: window.editorBase}))()"
+        )
+        .then((result = {}) => {
+          const getVersion = (base = "") => {
```
Comment (nemni8) [inline packages/corvid-cli/src/utils/electron.js]: simpler version of the function
const getVersion = (base = "") => base.substring(base.lastIndexOf('/') + 1)

File: packages/corvid-cli/src/apps/pull.js [hunk 2]
```diff
@@ -12,6 +12,9 @@ const pullApp = ({ useSsl = true, override = false, move = false } = {}) => ({
   serverMode: "clone",
   serverArgs: { override, move },
   handler: async (corvidConfig, win, client, localServerStatus) => {
+    logger.setExtraData("metasiteId", corvidConfig.metasiteId);
```
Comment (nemni8) [inline packages/corvid-cli/src/apps/pull.js]: logger.setExtraData("metasiteId", corvidConfig.metasiteId);
logger.setUserId(process.env.USER_GUID);
logger.setTag("command", process.env.COMMAND_CONTEXT);

the same code as in clone and open-editor
* maybe should be a util function
* maybe should be a function inside the logger that will set all 3 parameters together

File: packages/corvid-cli/src/utils/electron.js (hunk 1 above)
Comment (kostagersh) [reply packages/corvid-cli/src/utils/electron.js]: done

File: packages/corvid-cli/src/apps/pull.js (hunk 2 above)
Comment (kostagersh) [reply packages/corvid-cli/src/apps/pull.js]: done

**Comment by dor-itzhaki** [comment]
@kostagersh 	please update the commit message

File: packages/corvid-cli/src/apps/clone.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const chalk = require("chalk");
 const normalize = require("normalize-url");
 const { URL } = require("url");
 const getMessage = require("../messages");
-const { UserError } = require("corvid-local-logger");
+const logger = require("corvid-local-logger");
```
Comment (dor-itzhaki) [inline packages/corvid-cli/src/apps/clone.js]: FYI, you can do:
const {logger, UserError} = require('.....')

File: packages/corvid-cli/src/utils/electron.js [hunk 4]
```diff
@@ -152,14 +153,27 @@ async function openWindow(app, windowOptions = {}) {
       setTimeout(resolve, isDevTools ? 1000 : 0);
     }).then(async () => {
       const corvidConfig = await readCorvidConfig(".");
-
+      win.webContents
+        .executeJavaScript(
+          "(() => ({santaBase: window.santaBase, editorBase: window.editorBase}))()"
```
Comment (dor-itzhaki) [inline packages/corvid-cli/src/utils/electron.js]: please verify that this is correct when passing a specific version or localhost in a query parameter

File: packages/corvid-cli/src/utils/electron.js [hunk 5]
```diff
@@ -152,14 +153,27 @@ async function openWindow(app, windowOptions = {}) {
...
+        .then((result = {}) => {
+          const getVersion = (base = "") =>
+            base.substring(base.lastIndexOf("/") + 1);
+          const santaVersion = getVersion(get_(result, "santaBase"));
+          const editorVersion = getVersion(get_(result, "editorBase"));
+          logger.setExtraData("santa", santaVersion);
+          logger.setExtraData("editor", editorVersion);
+        })
+        .catch(e =>
+          logger.warn(`error executing js script in electron: ${e.message}`)
```
Comment (dor-itzhaki) [inline packages/corvid-cli/src/utils/electron.js]: "error reading editor versions"

File: packages/corvid-cli/src/apps/clone.js (hunk 3 above)
Comment (kostagersh) [reply packages/corvid-cli/src/apps/clone.js]: done

File: packages/corvid-cli/src/commands/login.js [hunk 6]
```diff
@@ -102,11 +105,15 @@ module.exports = {
   command: "login",
   describe: getMessage("Login_Command_Description"),
   handler: async args => {
+    process.env.COMMAND_CONTEXT = "login";
```
[the remark this answers is not in the document]
Comment (kostagersh) [reply packages/corvid-cli/src/commands/login.js]: done

File: packages/corvid-cli/src/utils/electron.js (hunk 4 above)
Comment (kostagersh) [reply packages/corvid-cli/src/utils/electron.js]: done

Comment (kostagersh) [reply packages/corvid-cli/src/utils/electron.js]: `santaBase
"https://localhost/"`

File: packages/corvid-cli/src/utils/electron.js (hunk 5 above)
Comment (kostagersh) [reply packages/corvid-cli/src/utils/electron.js]: done

File: packages/corvid-cli/src/utils/electron.js [hunk 7]
```diff
@@ -152,14 +153,25 @@ async function openWindow(app, windowOptions = {}) {
       setTimeout(resolve, isDevTools ? 1000 : 0);
     }).then(async () => {
       const corvidConfig = await readCorvidConfig(".");
-
+      win.webContents
+        .executeJavaScript(
+          "(() => ({santaBase: window.santaBase, editorBase: window.editorBase}))()"
+        )
+        .then((result = {}) => {
+          const getVersion = (base = "") =>
```
Comment (kostagersh) [inline packages/corvid-cli/src/utils/electron.js]: if domain is localhost write "local"

File: packages/corvid-cli/src/cli.js [hunk 8]
```diff
@@ -12,6 +13,8 @@ console.log(chalk.yellow(getMessage("Cli_Description_Yellow")));
 console.log(getMessage("Cli_Description"));
 
 logger.info(`running [${process.argv.slice(2).join(" ")}]`);
+process.env.COMMAND_CONTEXT = process.env.COMMAND_CONTEXT || process.argv[2];
```
Comment (kostagersh) [inline packages/corvid-cli/src/cli.js]: move all env settings to logger

Comment (kostagersh) [reply packages/corvid-cli/src/cli.js]: done

File: packages/corvid-cli/src/utils/electron.js (hunk 7 above)
Comment (kostagersh) [reply packages/corvid-cli/src/utils/electron.js]: done

File: packages/corvid-cli/src/commands/login.js [hunk 9]
```diff
@@ -92,7 +92,9 @@ async function loginCommand(spinner, args = {}) {
       ? messages.filter(({ msg }) => msg === "authCookie")
       : [];
     const authCookie = _.get(cookieMessages, [0, "cookie"]);
-    await sessionData.set({ uuid: parseSessionCookie(authCookie).userGuid });
+    const userId = parseSessionCookie(authCookie).userGuid;
+    logger.addSessionData({ userId });
```
Comment (dor-itzhaki) [inline packages/corvid-cli/src/commands/login.js]: can be moved to cli using sessionData.on

File: packages/corvid-cli/src/utils/electron.js [hunk 10]
```diff
@@ -152,14 +153,29 @@ async function openWindow(app, windowOptions = {}) {
       setTimeout(resolve, isDevTools ? 1000 : 0);
     }).then(async () => {
       const corvidConfig = await readCorvidConfig(".");
-
+      win.webContents
```
Comment (dor-itzhaki) [inline packages/corvid-cli/src/utils/electron.js]: please extract to a function:
const {santaVersion, editorVersion} = await getSantaVersions(win)
logger.addSessionData({...})

**Pull request merged**